### DISCUSSION TITLE
Embedding nuscript

### DISCUSSION THREAD

**Discussion by N4tus** [nushell/nushell#7782] [Ideas]
I was wondering if it is possible to embed nu as scripting language in other applications. Allowing the host to define the commands that are usable within a script and getting a REPL "for free".

And also if this is even something that the maintainers of nushell even want to support in the far future.

**Comment by fdncred** [comment]
Yes, this is exactly what [couchbase-shell](https://github.com/couchbaselabs/couchbase-shell) does. All the crates are published on crates.io so they're available for use.
